### PR TITLE
Raise error in has_session_key

### PR DESCRIPTION
**Pull request by jonathanudd**
Raising a error should end up in the action result right?
So maybe this is enough instead of https://github.com/dwellir-public/polkadot-operator/pull/45

I don't remember how and where I could test this though....

### CODE REVIEW

**Comment by Maharacha** [comment]
maybe this?
`jra polkadot/0 is-validating-next-era address=DuLr6CeLXezrfumF6EkqLeAx9paMcADYU6zHpSZVB8gvjht`
on `dwellir-kusama-validator-1  site-tunis`
```
Traceback (most recent call last):
  File "/var/lib/juju/agents/unit-polkadot-0/charm/./src/charm.py", line 308, in <module>
    ops.main(PolkadotCharm)
  File "/var/lib/juju/agents/unit-polkadot-0/charm/venv/ops/main.py", line 451, in __call__
    return main(charm_class, use_juju_for_storage=use_juju_for_storage)
  File "/var/lib/juju/agents/unit-polkadot-0/charm/venv/ops/main.py", line 436, in main
    _emit_charm_event(charm, dispatcher.event_name)
  File "/var/lib/juju/agents/unit-polkadot-0/charm/venv/ops/main.py", line 144, in _emit_charm_event
    event_to_emit.emit(*args, **kwargs)
  File "/var/lib/juju/agents/unit-polkadot-0/charm/venv/ops/framework.py", line 351, in emit
    framework._emit(event)
  File "/var/lib/juju/agents/unit-polkadot-0/charm/venv/ops/framework.py", line 853, in _emit
    self._reemit(event_path)
  File "/var/lib/juju/agents/unit-polkadot-0/charm/venv/ops/framework.py", line 942, in _reemit
    custom_handler(event)
  File "/var/lib/juju/agents/unit-polkadot-0/charm/./src/charm.py", line 253, in _on_is_validating_next_era_action
    event.set_results(results={'session_key': session_key})
  File "/var/lib/juju/agents/unit-polkadot-0/charm/venv/ops/charm.py", line 188, in set_results
    self.framework.model._backend.action_set(results)
  File "/var/lib/juju/agents/unit-polkadot-0/charm/venv/ops/model.py", line 3197, in action_set
    flat_results = _format_action_result_dict(results)
  File "/var/lib/juju/agents/unit-polkadot-0/charm/venv/ops/model.py", line 2890, in _format_action_result_dict
    raise ValueError("key '{!r}' is invalid: must be similar to 'key', 'some-key2', or "
ValueError: key ''session_key'' is invalid: must be similar to 'key', 'some-key2', or 'some.key'
```

File: src/polkadot_rpc_wrapper.py [hunk 1]
```diff
@@ -73,7 +73,7 @@ def get_system_peers(self) -> Tuple[list, bool]:
         peer_list = response_json['result']
         return peer_list, True
 
-    def has_session_key(self, session_key):
+    def has_session_key(self, session_key) -> bool:
```
Comment (Maharacha) [inline src/polkadot_rpc_wrapper.py:76]: It does not return a bool right?

Comment (Maharacha) [reply src/polkadot_rpc_wrapper.py:76]: Ah you are probably right
https://polkadot.js.org/docs/kusama/rpc/#hassessionkeyssessionkeys-bytes-bool

**Review by Maharacha — APPROVED** [review]

**Review by jakobilobi — COMMENTED** [review]
With this new `raise`, maybe make `_on_has_session_key_action` except that and fail the event?

**Comment by Maharacha** [comment]
> > With this new raise, maybe make _on_has_session_key_action except that and fail the event?
> 
> Will do.
> 
> I still don't know how to test this since the issue @Maharacha had was solved by an other cause. I also feel like I don't have the time to spend on this so we can at least skip it for the next revision.

Yeah close the PR and we'll handle it if problems arise.

**Pull request closed without merging**